docs/RFCs/0002-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,169 @@
+- `source_type`:
+    The string "crashtracking".
+- `stack`:
+    This represents the stack of the crashing thread.
+    See below for more details on how stacktraces are formatted.
+
+### Optional fields
+Any field not listed as "Required" is optional.
+In order to minimize logging overhead, producers SHOULD NOT emit anything for an optional field.
+Consumers MUST accept json with elided optional fields.
+
+- `counters`:
+    The crashtracker offers a mechanism for programs to register counters to track which operations were active at the time of the crash.
+    At present, this is only used by the profiler, but this may be extended in the future.
+- `files`:
+    The collector MAY collect useful files, such as `/proc/self/maps` or `/proc/meminfo`, and include them here.
+    Files are stored as a `Map<filename, contents>` where `contents` is an array of plain text strings, one per line.
+- `proc_info`: 
+    Currently, just tracks the PID of the crashing process.  
+    In the future, this may record additional info about the crashing process.
+- `siginfo`:
+    The name and signal number of the crashing signal (on UNIX systems)
+- `span_ids`: 
+    A vector of string identifiers, representing the active span ids at the time of program crash.
+    The collector SHOULD collect as many as it can, but MAY cap the number of spans that it tracks.
+- `trace_ids:`
+    A vector of string identifiers, representing the active span ids at the time of program crash.
+    The collector SHOULD collect as many as it can, but MAY cap the number of spans that it tracks.
+
+### Extensibility
+Future versions of the crashtracker MAY add additional fields.
+Parsers MUST accept unexpected optional fields, either by ignoring them, or by displaying them as additional data.
+The version number SHOULD be incremented for important optional fields, and MUST be incremented when a required field is added or removed.
+
+### Stacktraces
+Different languages and language runtimes have different representations of a stacktrace.
+The representation below attempts to collect as much information as possible.
+In addition, not all information may be available at crash-time on a given machine.
+For example, some libraries may have been shipped with debug symbols stripped, meaning that the only information available about a given frame may be the instruction pointer (`ip`) address, stored as a hex number "0xDEADBEEF".
+This address may be given as an absolute address, or a `NormalizedAddress`, which can be used by backend symbolication.
+We follow the [blazezym](https://github.com/libbpf/blazesym) format for normalized addresses.
+
+A stacktrace consists of 
+  - `format`:
+    An identifier describing the format of the stack trace.
+    Allows for extensibility to support different stack trace formats.
+    The format described below is identfied using the string "CrashTrackerV1"
+  - `frames`:
+    An array of `StackFrame`, described below
+
+#### StackFrame
```

Comment:
I'd like to request some data about the actual functions. Especially in C code there's information to be had "this variable was a null pointer - this looks like freed memory - probably valid address".

Reply:
You mean like the actual values of local variables on the stack?